### PR TITLE
feat(Model): fetch tableName from (SSM) Parameter Store

### PR DESCRIPTION
**Pull request by SajidHamza9**

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
     "rootDir": ".",
     "outDir": "./dist",
     "moduleResolution": "node",
-    "module": "commonjs",
+    "module": "ES2022",
```
Comment (MarioArnt) [inline tsconfig.json:7]: We want to keep support of previous NodeJS versions, please revert to CommonJS

File: src/base-model.ts [hunk 2]
```diff
@@ -371,13 +414,13 @@ export default abstract class Model<T> {
    * @param options: Additional options supported by AWS document client.
    * @returns  The scanned items (in the 1MB single scan operation limit) and the last evaluated key
    */
-  public scan(options?: Partial<ScanCommandInput>): Scan<T> {
+  public async scan(options?: Partial<ScanCommandInput>): Promise<Scan<T>> {
```
Comment (MarioArnt) [inline src/base-model.ts:417]: For backward compatibility reason we do not want to change public API and make `scan` and `query` method async.

Solution here is to leave TableName empty and resolve it when async method `count`, `exec` or `execAll` are called.

File: src/base-model.ts [hunk 3]
```diff
@@ -31,8 +32,50 @@ type Keys = SimpleKey[] | CompositeKey[];
...
+    set: setter
+  });
+}
+
+
 export default abstract class Model<T> {
-  protected tableName: string | undefined;
+
+  @SSMParam
+  protected tableName: string | Promise<string | undefined> | undefined;
```
Comment (MarioArnt) [inline src/base-model.ts:78]: Nice use case for a decorator 💯

**Review by MarioArnt — CHANGES_REQUESTED** [review]
Althrough tableName became now a possible Promise, we do not need to change public API here.
It would need a major release and rewrite A LOT of tests.
Have a look on PR #288 , the implementation is non-breaking and test are passing.
Maybe you can merge both of PRs good ideas: decarator-style and defer-resolve table name in query and scan.

**Review by ThibaRu — APPROVED** [review]

**Review by MarioArnt — CHANGES_REQUESTED** [review]
Well done !

Almost good for me 👍 

I'm just asking one more thing: 
Can you update README.md (first section) and mention this is now possible to load table name from SSM providing a code example.

**Review by MarioArnt — APPROVED** [review]

**Pull request merged**